include/extractor/guidance/turn_handler.hpp

```diff
@@ -45,9 +47,24 @@ class TurnHandler : public IntersectionHandler
                             Intersection intersection) const override final;
 
   private:
+    struct Fork
+    {
+        Intersection::iterator right;
+        Intersection::iterator left;
```

Comment:
since we are using iterators, wouldn't it be better to follow the stdlib paradigm for `begin`, `end` here? I'd prefer a begin / end approach at least, but given that `Fork` is wrapped into an optional, I'd be alright with `left`/`right`. However, `left` and `right` requires implicit knowledge about the order. So we could maybe use `ccw_right` and `ccw_left` to indicate the order here as well? This way you know which direction to iterate over.